crates/proto/src/dnssec/rdata/nsec3.rs

```diff
@@ -406,6 +420,44 @@ impl fmt::Display for NSEC3 {
     }
 }
 
+/// Helper struct used in deserialization of [`NSEC3`].
```

Comment:
Maybe expand on how this deals with `next_hashed_owner_name_base32` a little in a comment, to make it more obvious how/why it is different from the original?